src/algebra/big_operators/basic.lean

```diff
@@ -1706,6 +1706,14 @@ begin
     exact add_eq_union_left_of_le (finset.sup_le (λ x hx, le_sum_of_mem (mem_map_of_mem f hx))), },
 end
 
+lemma sup_powerset_len {α : Type*} [decidable_eq α] (x : multiset α) :
+  finset.sup (finset.range (x.card + 1)) (λ k, x.powerset_len k) = x.powerset :=
+ begin
+  convert bind_powerset_len x,
+  rw [multiset.bind, multiset.join, ←finset.range_coe, ←finset.sum_eq_multiset_sum],
+  exact eq_comm.mp (finset_sum_eq_sup_iff_disjoint.mpr (λ _ _ _ _ h, disjoint_powerset_len x h)),
```

Comment:
```suggestion
  exact (finset_sum_eq_sup_iff_disjoint.mpr (λ _ _ _ _ h, disjoint_powerset_len x h)).symm,
```

Reply:
Yeah, I tried that but it does not work:
```lean
type mismatch at application
  ?m_2.disjoint_powerset_len h
term
  h
has type
  _x ≠ _x_1
but is expected to have type
  ?m_1 ≠ ?m_1
```
I think the matching in ```finset_sum_eq_sup_iff_disjoint.mpr (λ _ _ _ _ h, disjoint_powerset_len _ h)``` is somehow messed up. I am not sure why ```eq_comm.mp``` works then, but it does...

Reply:
I would guess this works?
```suggestion
  exact eq.symm (finset_sum_eq_sup_iff_disjoint.mpr (λ _ _ _ _ h, disjoint_powerset_len x h)),
```

Reply:
Yes, that works. I am not sure why ```eq.symm``` works but not ```.symm``` though